sup3r/postprocessing/collection.py

```diff
@@ -229,7 +230,70 @@ def get_data(self, file_path, feature, time_index, meta, scale_factor,
         self.data[row_slice, col_slice] = f_data
 
     @staticmethod
-    def _get_collection_attrs(file_paths, feature, sort=True, sort_key=None):
+    def _get_file_attrs(file):
+        """Get meta data and time index for a single file"""
+        with RexOutputs(file, mode='r') as f:
+            meta = f.meta
+            time_index = f.time_index
+        return meta, time_index
+
+    @classmethod
+    def _get_collection_attrs_parallel(cls, file_paths, max_workers=None):
+        """Get meta data and time index from a file list to be collected.
+
+        Assumes the file list is chunked in time (row chunked).
+
+        Parameters
+        ----------
+        file_paths : list | str
+            Explicit list of str file paths that will be sorted and collected
+            or a single string with unix-style /search/patt*ern.h5. Files
+            should have non-overlapping time_index dataset and fully
+            overlapping meta dataset.
+        max_workers : int | None
+            Number of workers to use in parallel. 1 runs serial,
+            None will use all available workers.
+
+        Returns
+        -------
+        time_index : pd.datetimeindex
+            List of datetime indices for each file that is being collected
+        meta : pd.DataFrame
+            List of meta data for each files that is being
+            collected
+        """
+
+        time_index = [None] * len(file_paths)
+        meta = [None] * len(file_paths)
+        futures = {}
+        with ThreadPoolExecutor(max_workers=max_workers) as exe:
+            for i, fn in enumerate(file_paths):
+                future = exe.submit(cls._get_file_attrs, fn)
+                futures[future] = i
+
+            interval = np.int(np.ceil(len(futures) / 10))
+            for i, future in enumerate(as_completed(futures)):
+                if interval > 0 and i % interval == 0:
+                    mem = psutil.virtual_memory()
+                    logger.info('Collection futures completed: '
```

Comment:
This should not be called a collection future. Will be confused with the actual collection futures.